Userland/Libraries/LibJS/SyntaxHighlighter.cpp

```diff
@@ -92,16 +94,39 @@ void SyntaxHighlighter::rehighlight(Palette const& palette)
             span.range.end().line(), span.range.end().column());
     };
 
+    struct TokenData {
+        Token token;
+        GUI::TextRange range;
+    };
+    Vector<TokenData> folding_region_start_tokens;
+
     bool was_eof = false;
     for (auto token = lexer.next(); !was_eof; token = lexer.next()) {
         append_token(token.trivia(), token, true);
+
+        auto token_start_position = position;
         append_token(token.value(), token, false);
 
         if (token.type() == TokenType::Eof)
             was_eof = true;
+
+        // Create folding regions for {} blocks
```

Comment:
Should we just do these wholesale based on indentation in all files that don't come with their own folding rules?

some :canned_food: for thought

Reply:
:thonk: That might be interesting... I'll make a note to try it.